app/assets/javascripts/discourse/app/controllers/signup.js

```diff
@@ -453,7 +453,7 @@ export default class SignupPageController extends Controller {
     const destinationUrl = this.authOptions?.destination_url;
 
     if (!isEmpty(destinationUrl)) {
-      cookie("destination_url", destinationUrl, { path: "/" });
+      cookie("destination_url", destinationUrl);
```

Comment:
`{ path: "/" }` is the default.

Reply:
Can we also ensure that destinationUrl is subfolder aware? 

Unless there's a piece of logic somewhere else modifying the cookie, it doesn't *seem* like it takes subfolders into account:

https://github.com/discourse/discourse/blob/a9cf9aeba1796d718f658bc5a37035251dd9dce0/app/assets/javascripts/discourse/app/controllers/login.js#L155-L162

Reply:
What makes you think the behavior changed when it comes to handling subfolder? 🤔

Reply:
I think it's unrelated to the code path and current issue at hand, but following the way the cookie set for this passkey function, it appears like window location is pulling `destination_url` directly which can be set to `/`.

Just wanting to call out here is that subfolder is being taken into account for this issue as well.

The cookie interacting with the cookie isn't blocking the current PR here, but probably good as a followup.